suite-native/notifications/src/components/TransactionNotification.tsx

```diff
@@ -1,92 +1,122 @@
+import { useSelector, useDispatch } from 'react-redux';
 
 import { useNavigation } from '@react-navigation/native';
 
-import { AccountKey, TransactionType } from '@suite-common/wallet-types';
+import { TransactionType } from '@suite-common/wallet-types';
 import { TransactionIcon } from '@suite-native/transactions';
 import {
     TransactionsRootState,
-    selectTransactionByTxidAndAccountKey,
     selectTransactionFirstTargetAddress,
+    AccountsRootState,
+    selectAccountKeyByDescriptorAndNetworkSymbol,
 } from '@suite-common/wallet-core';
 import {
     RootStackRoutes,
     StackNavigationProps,
     RootStackParamList,
 } from '@suite-native/navigation';
 import { Icon } from '@trezor/icons';
+import {
+    EventNotification,
+    notificationsActions,
+    TransactionEventNotification,
+} from '@suite-common/toast-notifications';
 
 import { Notification } from './Notification';
 import { TransactionNotificationDescription } from './TransactionNotificationDescription';
 
 type TransactionNotificationProps = {
-    txid: string;
-    accountKey: AccountKey;
+    notification: TransactionEventNotification;
     isHiddenAutomatically?: boolean;
 };
 
 type TransactionTypeProperties = {
     title?: string;
     prefix?: string;
+    transactionType: TransactionType;
+    isIconAnimated: boolean;
 };
 
 const transactionTypeToContentMap = {
-    recv: { title: 'Incoming transaction', prefix: 'from' },
-    sent: { title: 'Sending transaction', prefix: 'to' },
-    self: { title: 'Sending transaction', prefix: 'to' },
-    joint: { title: 'Sending transaction', prefix: 'to' },
-    failed: { title: 'Sending transaction', prefix: 'to' },
-    unknown: { title: 'Sending transaction', prefix: 'to' },
-} as const satisfies Record<TransactionType, TransactionTypeProperties>;
+    'tx-received': {
+        title: 'Incoming transaction',
+        prefix: 'from',
+        transactionType: 'recv',
+        isIconAnimated: true,
+    },
+    'tx-confirmed': {
+        title: 'Received transaction',
+        prefix: 'from',
+        transactionType: 'recv',
+        isIconAnimated: false,
+    },
+    'tx-sent': {
+        title: 'Sending transaction',
+        prefix: 'to',
+        transactionType: 'sent',
+        isIconAnimated: true,
+    },
+} as const satisfies Record<TransactionEventNotification['type'], TransactionTypeProperties>;
 
 export const TransactionNotification = ({
-    txid,
-    accountKey,
+    notification,
     isHiddenAutomatically = true,
 }: TransactionNotificationProps) => {
     const navigation =
         useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AppTabs>>();
+    const dispatch = useDispatch();
+    const accountKey =
+        useSelector((state: AccountsRootState) =>
+            selectAccountKeyByDescriptorAndNetworkSymbol(
+                state,
+                notification.descriptor,
+                notification.symbol,
+            ),
+        ) ?? '';
```

Comment:
You shouldn't do fallbacks like this in component. This should be handled in selector itself

Reply:
fixed  53e695678